Modules/ZDC/src/ZDCRawDataTask.cxx

```diff
@@ -0,0 +1,1032 @@
+#include <TROOT.h>
+#include <TPad.h>
+#include <TString.h>
+#include <TStyle.h>
+#include <TPaveStats.h>
+#include "CommonConstants/LHCConstants.h"
+#include "ZDCSimulation/Digits2Raw.h"
+#include "ZDCSimulation/ZDCSimParam.h"
+#include "FairLogger.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <ctype.h>
+#include <boost/algorithm/string.hpp>
+using namespace o2::zdc;
+
+namespace o2::quality_control_modules::zdc
+{
+
+ZDCRawDataTask::~ZDCRawDataTask()
+{
+  delete mHistogram;
+}
+
+void ZDCRawDataTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info, Support) << "initialize ZDCRawDataTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
+  if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
+    ILOG(Info, Devel) << "Custom parameter - myOwnKey: " << param->second << ENDM;
+  }
+  init();
+  mHistogram = new TH1F("eventSize", "eventSize", 20, 0, 30000);
+  getObjectsManager()->startPublishing(mHistogram);
+  getObjectsManager()->startPublishing(new TH1F("example2", "example2", 20, 0, 30000));
+  try {
+    getObjectsManager()->addMetadata(mHistogram->GetName(), "custom", "34");
+  } catch (...) {
+    // some methods can throw exceptions, it is indicated in their doxygen.
+    // In case it is recoverable, it is recommended to catch them and do something meaningful.
+    // Here we don't care that the metadata was not added and just log the event.
+    ILOG(Warning, Support) << "Metadata could not be added to " << mHistogram->GetName() << ENDM;
+  }
+}
+
+void ZDCRawDataTask::startOfActivity(Activity& activity)
+{
+  ILOG(Info, Support) << "startOfActivity" << activity.mId << ENDM;
+  mHistogram->Reset();
+}
+
+void ZDCRawDataTask::startOfCycle()
+{
+  ILOG(Info, Support) << "startOfCycle" << ENDM;
+}
+
+void ZDCRawDataTask::monitorData(o2::framework::ProcessingContext& ctx){
+
+    o2::framework::InputRecord &inputs = ctx.inputs();
+    o2::framework::DPLRawParser parser(inputs);
+    for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
+        auto const* payload = it.data();
+        size_t payloadSize = it.size();
+        if (payload != nullptr) {
+            mHistogram->Fill((double)payloadSize);
+            for (int32_t ip = 0; ip < (int32_t)payloadSize; ip += 16) {
+                processWord((const uint32_t*)&payload[ip]);
+            }
+       }
+    }
+}
+
+void ZDCRawDataTask::endOfCycle()
+{
+  ILOG(Info, Support) << "endOfCycle" << ENDM;
+}
+
+void ZDCRawDataTask::endOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "endOfActivity" << ENDM;
+}
+
+void ZDCRawDataTask::reset()
+{
+  // clean all the monitor objects here
+
+  ILOG(Info, Support) << "Resetting the histogram" << ENDM;
+  mHistogram->Reset();
```

Comment:
Please reset all the monitor objects here. If you don't, several functionalities will not work - e.g. merging, moving windows, start->stop->start sequences.

Reply:
Ok I'll do it